var/spack/repos/builtin/packages/kaldi/package.py

```diff
@@ -0,0 +1,91 @@
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+from distutils.dir_util import copy_tree
+
+
+class Kaldi(Package):
+    """Kaldi is a toolkit for speech recognition written 
+    in C++ and licensed under the Apache License v2.0. 
+    Kaldi is intended for use by speech recognition researchers."""
+
+    homepage = "https://github.com/kaldi-asr/kaldi"
+    url      = "https://github.com/kaldi-asr/kaldi/archive/master.zip"
+
+    version('master', git='https://github.com/kaldi-asr/kaldi.git')
+
+    variant('shared', default=True,
+            description='build shared libraries')
+    variant('double', default=False,
+            description='build with double precision floats')
+    variant('cuda', default=False,
+            description='build with CUDA')
+
+    depends_on('blas')
+    depends_on('speex')
+    depends_on('openfst@1.6:')
+    depends_on('cuda', when='+cuda')
+    depends_on('sph2pipe', type='run')
+    depends_on('sctk', type='run')
+
+    def install(self, spec, prefix):
+        configure_args = [
+            '--threaded-math',
+            '--speex-root=' + spec['speex'].prefix,
+            '--fst-root=' + spec['openfst'].prefix,
+            '--fst-version=' + str(spec['openfst'].version)
+        ]
+    
+        if '~shared' in spec:
+            configure_args.append('--static')
+        else:
+            configure_args.append('--shared')
+
+        if '^openblas' in spec:
+            configure_args.append('--openblas-root=' + spec['blas'].prefix)
+            configure_args.append('--mathlib=OPENBLAS')
+        elif '^atlas' in spec:
+            configure_args.append('--atlas-root=' + spec['blas'].prefix)
+            configure_args.append('--mathlib=ATLAS')
+        elif '^intel-parallel-studio' in spec or '^intel-mkl' in spec:
+            configure_args.append('--mkl-root=' + spec['blas'].prefix)
+            configure_args.append('--mathlib=MKL')
```

Comment:
@weijianwen I would add extra ifs to set correctly `--threaded-atlas` (atlas+pthread) or `--threaded-math` (opneblas+openmp) or `--mkl-threading` (intel-mkl+openmp). 

As I said about, the problem here is that `kaldi`'s config system does not allow users to explicitly give a list of libs. It tries to be smart it pick up everything itself, which is good, but it would be even better to have full control.